core/primitives/src/utils.rs

```diff
@@ -276,149 +264,76 @@ impl KeyForData {
         })?))
     }
 
-    pub fn parse_data_key<'a>(
-        raw_key: &'a [u8],
-        account_id: &AccountId,
-    ) -> Result<&'a [u8], std::io::Error> {
-        let prefix_len = Self::estimate_len(account_id, &[]);
-        if raw_key.len() < prefix_len {
-            return Err(std::io::Error::new(
+    pub fn parse_account_id_from_account_key(raw_key: &[u8]) -> Result<AccountId, std::io::Error> {
+        let account_id = parse_account_id_prefix(col::ACCOUNT, raw_key)?;
+        Ok(AccountId::from(std::str::from_utf8(account_id).map_err(|_| {
+            std::io::Error::new(
                 std::io::ErrorKind::InvalidData,
-                "raw key is too short for KeyForData",
-            ));
-        }
-        Ok(&raw_key[prefix_len..])
+                "raw key does not have a valid AccountId to be KeyForAccount",
+            )
+        })?))
     }
-}
-
-#[derive(derive_more::AsRef, derive_more::Into)]
-#[as_ref(forward)]
-pub struct KeyForContractCode(Vec<u8>);
-
-impl TrieKey for KeyForContractCode {}
 
-impl KeyForContractCode {
-    pub fn new(account_id: &AccountId) -> Self {
-        Self(KeyForColumnAccountId::with_capacity(col::CODE, account_id, 0).into())
+    pub fn parse_account_id_from_access_key_key(
+        raw_key: &[u8],
+    ) -> Result<AccountId, std::io::Error> {
+        let account_id_prefix = parse_account_id_prefix(col::ACCESS_KEY, raw_key)?;
+        let public_key_position = if let Some(index) =
+            account_id_prefix.iter().enumerate().find(|(_, c)| **c == 2).map(|(index, _)| index)
```

Comment:
is there a way to avoid this hack `**c == 2`?

Reply:
All those functions within `mod trie_key_parsers` are temporary and should be removed once possible
It's moved from the deleted code. Didn't change it.

Reply:
@bowenwang1996 That is my hack :disappointed:, and as @evgenykuzyakov said, that should be fixed with #2244. I thought that would be refactored before anyone spot it (well, I didn't hide it: https://github.com/nearprotocol/nearcore/pull/2253#discussion_r392026033)